buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -108,25 +107,17 @@ class PerformanceTestPlugin : Plugin<Project> {
 
     private
     fun Project.registerForkPointDistributionTask() {
+        val determineForkPoint = tasks.register("determineForkPoint", DetermineForkPoint::class)
         val buildForkPointDistribution = tasks.register("buildForkPointDistribution", BuildForkPointDistribution::class) {
-            dependsOn("determineForkPoint")
-        }
-        tasks.register("determineForkPoint") {
-            doLast {
-                val masterForkPointCommit = execAndGetStdout("git", "merge-base", "origin/master", "HEAD")
-                val releaseForkPointCommit = execAndGetStdout("git", "merge-base", "origin/release", "HEAD")
-                val forkPointCommit =
-                    if (exec { isIgnoreExitValue = true; commandLine("git", "merge-base", "--is-ancestor", masterForkPointCommit, releaseForkPointCommit) }.exitValue == 0)
-                        releaseForkPointCommit
-                    else
-                        masterForkPointCommit
-                buildForkPointDistribution.get().determineForkPoint(execAndGetStdout("git", "rev-parse", "--short", forkPointCommit), execAndGetStdout("git", "show", "$forkPointCommit:version.txt"))
-            }
+            dependsOn(determineForkPoint)
         }
+
+        buildForkPointDistribution.get().forkPointDistributionVersion.set(determineForkPoint.get().forkPointDistributionVersion)
```

Comment:
No need to realize those two tasks. Just do this in the configure action just above:
```
forkPointDistributionVersion.set(determineForkPoint.flatMap { it.forkPointDistributionVersion })
```